### PR TITLE
Upgrade PHPMailer to 6.4.x

### PR DESCRIPTION
**Pull request by Dillon-Brown**
## Description

- Moved PHPMailer to composer.
- Upgrade PHPMailer to 6.4.x.
- Removed old phpmailer references and kept classes for BC.

## How To Test This

1. Regression test emails.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: modules/Emails/Email.php [hunk 1]
```diff
@@ -1278,7 +1278,7 @@ public function email2Send($request)
                     if (!empty($note->id)) {
                         $filename = $note->filename;
                         $noteGUID = preg_replace('/[^a-z0-9\-]/', "", $note->id);
-                        $fileLocation = "upload://{$noteGUID}";
+                        $fileLocation = "upload/{$noteGUID}";
```
Comment (clemente-raposo) [inline modules/Emails/Email.php:1281]: Hi @Dillon-Brown,

Why do we need this change?

Comment (Dillon-Brown) [reply modules/Emails/Email.php:1281]: It's for compatibility with PHPMailer 6.4.x. I've pulled in the change from: https://github.com/salesagility/SuiteCRM/pull/6938. Which was to fix https://github.com/salesagility/SuiteCRM/issues/6932 when we upgraded to 6.4.x in 7.11.x.

Comment (clemente-raposo) [reply modules/Emails/Email.php:1281]: ok, thanks Dillon

**Review by clemente-raposo — APPROVED** [review]
Changes look ok.

**Review by Mac-Rae — APPROVED** [review]

**Comment by Mac-Rae** [comment]
Assessed :+1:

**Comment by pgorod** [comment]
@Dillon-Brown @clemente-raposo I think we had issues a similar change before, though I don't fully remember the details. There is a configuration variable called [upload_dir](https://github.com/salesagility/SuiteCRM/search?q=upload_dir&type=) which lets people change the directory for uploads. 

Internally, I think that is implemented with something called "streams". Changing these **`upload://`** paths to just **`upload/`** is basically hard-coding the path again. In some places, not all, which can get a bit messy...

**Comment by Dillon-Brown** [comment]
> There is a configuration variable called upload_dir which lets people change the directory for uploads.

That's actually a really good point, I don't see how this would work if you had that configured unless it just doesn't apply to these files and they always go into a hardcoded paths. It's definitely something we'll need to double-check but as these changes have been in 7.11.x for quite a while I don't think it would be too high of a priority.

**Comment by pgorod** [comment]
I confess I have some trouble getting my head around these PHP streams concept, in the way that it is used for the uploads. But apparently it is registered with [stream_wrapper_register](https://www.php.net/manual/en/function.stream-wrapper-register.php) from here:

https://github.com/salesagility/SuiteCRM/blob/master/include/entryPoint.php#L141-L142
which calls 
https://github.com/salesagility/SuiteCRM/blob/master/include/UploadStream.php#L148

From that point on, I believe the expression **`upload://`** will refer our own "protocol" and invoke our [UploadStream class](https://github.com/salesagility/SuiteCRM/blob/master/include/UploadStream.php), which I suppose will use whatever directory is in `$GLOBALS['sugar_config']['upload_dir']`.

So the sneaky part is that when we're calling regular PHP file functions, that `UploadStream` class is actually intervening, if the filename happens to use the `upload::/` protocol.

Honestly, I think we should just leave this as it was. Exactly which compatibility problem were you getting with PHP Mailer?

Another way to remove the protocol but keep the configurability is to use some construct like this:
`$location = $GLOBALS['sugar_config']['upload_dir']."/" . $filename;`

We actually find constructs like this all over the code, which happens whenever the developer wasn't understanding how to actually use the Streams... I've done this mistake myself in my code...

**Comment by Dillon-Brown** [comment]
> Honestly, I think we should just leave this as it was. Exactly which compatibility problem were you getting with PHP Mailer?

@pgorod So really all I was doing here was backporting the PHPMailer 6.4.x upgrade from 7.11.x to 7.10.x. The extra commit which changes the ```upload://``` I pulled in from here: https://github.com/salesagility/SuiteCRM/pull/6938. Which was a fix intended to fix an issue raised when we did the 6.4.x upgrade on 7.11.x here: https://github.com/salesagility/SuiteCRM/issues/6932.

**Comment by pgorod** [comment]
Ok
so I guess the correct fix for this would be to resolve to a real path before sending it into the PHPMailer package.

There's a method in **UploadStream.php** which appears to do just this:

```php
    /**
     * Get real FS path of the upload stream file
     * @param string $path Upload stream path (with upload://)
     * @return string FS path
     */
    public static function path($path)
    {
        $path = substr($path, strlen(self::STREAM_NAME) + 3); // cut off upload://
        $path = str_replace("\\", "/", $path); // canonicalize path
        if ($path == ".." || substr($path, 0, 3) == "../" || substr($path, -3, 3) == "/.." || strstr($path, "/../")) {
            return null;
        }

        return self::getDir() . "/" . $path;
    }

```

This would leave the configurable `upload_dir` system intact.

**Pull request merged**